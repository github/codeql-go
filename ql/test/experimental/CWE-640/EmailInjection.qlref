experimental/CWE-640/EmailInjection.ql