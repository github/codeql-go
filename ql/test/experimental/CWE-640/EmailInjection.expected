edges
| email.go:22:10:22:17 | selection of Header : Header | email.go:25:56:25:67 | type conversion |
| email.go:32:21:32:31 | call to Referer : string | email.go:34:57:34:78 | type conversion |
| email.go:40:21:40:31 | call to Referer : string | email.go:44:25:44:38 | untrustedInput |
| email.go:49:21:49:31 | call to Referer : string | email.go:55:46:55:59 | untrustedInput |
| email.go:49:21:49:31 | call to Referer : string | email.go:56:52:56:65 | untrustedInput |
| email.go:61:21:61:31 | call to Referer : string | email.go:63:47:63:60 | untrustedInput |
| email.go:71:21:71:31 | call to Referer : string | email.go:77:47:77:60 | untrustedInput |
| email.go:85:21:85:31 | call to Referer : string | email.go:92:37:92:50 | untrustedInput |
| email.go:85:21:85:31 | call to Referer : string | email.go:94:48:94:61 | untrustedInput |
nodes
| email.go:22:10:22:17 | selection of Header : Header | semmle.label | selection of Header : Header |
| email.go:25:56:25:67 | type conversion | semmle.label | type conversion |
| email.go:32:21:32:31 | call to Referer : string | semmle.label | call to Referer : string |
| email.go:34:57:34:78 | type conversion | semmle.label | type conversion |
| email.go:40:21:40:31 | call to Referer : string | semmle.label | call to Referer : string |
| email.go:44:25:44:38 | untrustedInput | semmle.label | untrustedInput |
| email.go:49:21:49:31 | call to Referer : string | semmle.label | call to Referer : string |
| email.go:55:46:55:59 | untrustedInput | semmle.label | untrustedInput |
| email.go:56:52:56:65 | untrustedInput | semmle.label | untrustedInput |
| email.go:61:21:61:31 | call to Referer : string | semmle.label | call to Referer : string |
| email.go:63:47:63:60 | untrustedInput | semmle.label | untrustedInput |
| email.go:71:21:71:31 | call to Referer : string | semmle.label | call to Referer : string |
| email.go:77:47:77:60 | untrustedInput | semmle.label | untrustedInput |
| email.go:85:21:85:31 | call to Referer : string | semmle.label | call to Referer : string |
| email.go:92:37:92:50 | untrustedInput | semmle.label | untrustedInput |
| email.go:94:48:94:61 | untrustedInput | semmle.label | untrustedInput |
#select
| email.go:25:56:25:67 | type conversion | email.go:22:10:22:17 | selection of Header : Header | email.go:25:56:25:67 | type conversion | Email content may contain $@. | email.go:22:10:22:17 | selection of Header | untrusted input |
| email.go:34:57:34:78 | type conversion | email.go:32:21:32:31 | call to Referer : string | email.go:34:57:34:78 | type conversion | Email content may contain $@. | email.go:32:21:32:31 | call to Referer | untrusted input |
| email.go:44:25:44:38 | untrustedInput | email.go:40:21:40:31 | call to Referer : string | email.go:44:25:44:38 | untrustedInput | Email content may contain $@. | email.go:40:21:40:31 | call to Referer | untrusted input |
| email.go:55:46:55:59 | untrustedInput | email.go:49:21:49:31 | call to Referer : string | email.go:55:46:55:59 | untrustedInput | Email content may contain $@. | email.go:49:21:49:31 | call to Referer | untrusted input |
| email.go:56:52:56:65 | untrustedInput | email.go:49:21:49:31 | call to Referer : string | email.go:56:52:56:65 | untrustedInput | Email content may contain $@. | email.go:49:21:49:31 | call to Referer | untrusted input |
| email.go:63:47:63:60 | untrustedInput | email.go:61:21:61:31 | call to Referer : string | email.go:63:47:63:60 | untrustedInput | Email content may contain $@. | email.go:61:21:61:31 | call to Referer | untrusted input |
| email.go:77:47:77:60 | untrustedInput | email.go:71:21:71:31 | call to Referer : string | email.go:77:47:77:60 | untrustedInput | Email content may contain $@. | email.go:71:21:71:31 | call to Referer | untrusted input |
| email.go:92:37:92:50 | untrustedInput | email.go:85:21:85:31 | call to Referer : string | email.go:92:37:92:50 | untrustedInput | Email content may contain $@. | email.go:85:21:85:31 | call to Referer | untrusted input |
| email.go:94:48:94:61 | untrustedInput | email.go:85:21:85:31 | call to Referer : string | email.go:94:48:94:61 | untrustedInput | Email content may contain $@. | email.go:85:21:85:31 | call to Referer | untrusted input |
